awx/ui_next/src/components/Search/Search.jsx

```diff
@@ -110,9 +110,9 @@ class Search extends React.Component {
     const { searchKey, searchValue } = this.state;
     const { onSearch, qsConfig } = this.props;
 
-    const isNonStringField = key =>
-      qsConfig.integerFields.filter(field => field === key).length ||
-      qsConfig.dateFields.filter(field => field === key).length;
+    const isNonStringField =
+      qsConfig.integerFields.filter(field => field === searchKey).length ||
+      qsConfig.dateFields.filter(field => field === searchKey).length;
 
     // TODO: this will probably become more sophisticated, where date
     // fields and string fields are passed to a formatter
```

Comment:
looks like you're still invoking `isNonStringField` as a function instead of testing its boolean value directly